src/game/services/Rules/Rules.test.ts

```diff
@@ -0,0 +1,110 @@
+import { stubPlayer } from '../../../test-utils/stubs/players'
+import { pumpkin } from '../../cards/crops/pumpkin'
+import { carrot } from '../../cards'
+import { deckSize, initialHandSize } from '../../config'
+import { isGame } from '../../types/guards'
+import { handlePlayFromHand as mockCropHandlePlayFromHand } from '../../cards/crops/handlePlayFromHand'
+import { IGame, IPlayer } from '../../types'
+
+import { Rules } from './Rules'
+
+const player1 = stubPlayer()
+const player2 = stubPlayer()
+
+jest.mock('../../cards/crops/handlePlayFromHand', () => {
+  return {
+    handlePlayFromHand: jest.fn(),
+  }
+})
+
+// Make player2's deck slightly different from player1's to prevent false
+// positives.
+player2.deck[deckSize - 1] = pumpkin.id
+
+describe('Rules', () => {
+  describe('processGameStart', () => {
+    test('creates a new game', () => {
+      const game = Rules.processGameStart([player1, player2])
+
+      expect(isGame(game)).toBe(true)
+    })
+
+    test('shuffles decks', () => {
+      const game = Rules.processGameStart([player1, player2])
+      const [player1Id, player2Id] = Object.keys(game.table.players)
+
+      expect(game.table.players[player1Id].deck).toEqual(
+        player1.deck.slice().reverse().slice(initialHandSize)
+      )
```

Comment:
i would probably do this expectation with a mock instead, something like `expect(shuffle).toHaveBeenCalledWith(player1.deck)`. since shuffle is ultimately mocked testing the output of it doesn't feel as useful as just making sure it's invoked with the expected argument